tests/integration_tests/device/cuda/navigation_validation.cu

```diff
@@ -0,0 +1,221 @@
+    static constexpr auto is_no_bfield{
+        std::is_same_v<bfield_t, navigation_validator::empty_bfield>};
+    using stepper_t =
+        std::conditional_t<is_no_bfield, line_stepper_t, rk_stepper_t>;
+
+    // Inspector that records all encountered surfaces
+    using intersection_t = typename intersection_record_t::intersection_type;
+    /*using object_tracer_t =
+        navigation::object_tracer<intersection_t, vecmem::device_vector,
+                                  navigation::status::e_on_module,
+                                  navigation::status::e_on_portal>;
+    // Navigation with inspection
+    using navigator_t = navigator<detector_device_t, object_tracer_t,
+    intersection_t>;*/
+    using navigator_t = navigator<detector_device_t>;
+
+    // Propagator with pathlimit aborter
+    using actor_chain_t = actor_chain<tuple, pathlimit_aborter>;
+    using propagator_t = propagator<stepper_t, navigator_t, actor_chain_t>;
+
+    static_assert(std::is_same_v<typename detector_t::view_type,
+                                 typename detector_device_t::view_type>,
+                  "Host and device detector views do not match");
+
+    detector_device_t det(det_data);
+
+    using mask_id = typename detector_device_t::masks::id;
+    using acc_id = typename detector_device_t::accel::id;
+    /*printf("Number of volumes: %d\n", det.volumes().size());
+    printf("Number of transforms: %d\n", det.transform_store().size());
+    printf("Number of rectangles: %d\n",
+           det.mask_store().template get<mask_id::e_rectangle2>().size());
+    printf("Number of trapezoids: %d\n",
+           det.mask_store().template get<mask_id::e_trapezoid2>().size());
+    printf("Number of portal discs: %d\n",
+           det.mask_store().template get<mask_id::e_portal_ring2>().size());
+    printf("Number of portal cylinders: %d\n",
+           det.mask_store().template get<mask_id::e_portal_cylinder2>().size());
+    printf("Number of portal collections: %d\n",
+           det.accelerator_store().template
+    get<acc_id::e_brute_force>().size()); printf("Number of disc grids: %d\n",
+           det.accelerator_store().template get<acc_id::e_disc_grid>().size());
+    printf("Number of cylinder grids: %d\n",
+           det.accelerator_store().template
+    get<acc_id::e_cylinder2_grid>().size());*/
+
+    vecmem::jagged_device_vector<intersection_t> candidates(candidates_data);
+    vecmem::jagged_device_vector<const intersection_record_t>
+        intersection_traces(intersection_traces_view);
+    vecmem::jagged_device_vector<intersection_t> intersection_records(
+        intersection_records_view);
+
+    assert(candidates.size() > 0);
+    assert(intersection_traces.size() == intersection_records.size());
+
+    for (unsigned int i = 0u; i < candidates.size(); ++i) {
+        // printf("cache capacity: %u\n", candidates.at(i).capacity());
+        assert(candidates.at(i).capacity() > 0);
+    }
+
```

Comment:
So... which one of these assertions fails for you at the moment?

Reply:
The one in l.94. I have a subvector per track, but that subvector has zero capacity. I am not sure the buffer actually needs to be resizable or what that means precisely?